### PR TITLE
Add alerter for IRIS IRP system

### PR DESCRIPTION
**Pull request by malinkinsa**
## Description

Added a new alerter type for the ability to create alerts and cases in the IRIS IRP system.

## Checklist

- [x] I have reviewed the [contributing guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).
- [x] I have included unit tests for my changes or additions.
- [x] I have successfully run `make test-docker` with my changes.
- [x] I have manually tested all relevant modes of the change in this PR.
- [x] I have updated the [documentation](https://elastalert2.readthedocs.io).
- [x] I have updated the [changelog](https://github.com/jertel/elastalert2/blob/master/CHANGELOG.md).

### CODE REVIEW

File: docs/source/ruletypes.rst [hunk 1]
```diff
@@ -2579,6 +2579,66 @@ Example usage with json string formatting::
...
+
+``iris_alert_tags``: List of tags.
+
+``iris_alert_status_id``: Alert status. Can be: ``1 - Unspecified``, ``2 - New``, ``3 - Assigned``, ``4 - In progress``, ``5 - Pending``, ``6 - Closed``, ``7 - Merged``. The default value is: `2`.
+
+``iris_alert_source_link``: Link, if needed.
+
+``iris_alert_severity_id``: Alert severity. Can be: ``1 - Unspecified``, ``2 - Informational``, ``3 - Low``, ``4 - Medium``, ``5 - High``, ``6 - Critical``. The default value is: `1`.
+
+``iris_alert_context``:
```
Comment (jertel) [inline docs/source/ruletypes.rst:2618]: This needs a description. Make sure the expected type is clarified. I could see someone putting a string here which might cause some unexpected behavior with the new alerting code.

File: elastalert/loaders.py [hunk 2]
```diff
@@ -25,7 +25,8 @@
 import elastalert.alerters.googlechat
 import elastalert.alerters.httppost
 import elastalert.alerters.httppost2
-import elastalert.alerters.lark
+import elastalert.alerters.iris
+# import elastalert.alerters.lark
```
Comment (jertel) [inline elastalert/loaders.py:29]: Why is this lark import disabled?

File: elastalert/loaders.py [hunk 3]
```diff
@@ -127,12 +128,13 @@ class RulesLoader(object):
         'zabbix': ZabbixAlerter,
         'discord': elastalert.alerters.discord.DiscordAlerter,
         'dingtalk': elastalert.alerters.dingtalk.DingTalkAlerter,
-        'lark': elastalert.alerters.lark.LarkAlerter,
+        # 'lark': elastalert.alerters.lark.LarkAlerter,
```
Comment (jertel) [inline elastalert/loaders.py:131]: Same here with Lark being disabled.

**Review by jertel — CHANGES_REQUESTED** [review]
Thanks for this useful contribution! I added a few comments to this review. I'll enable the automated tests in the meantime.

**Comment by jertel** [comment]
The reason the automation failed is due to the linter finding some white space violations in your new code/tests.

**Comment by malinkinsa** [comment]
I see a problem in the tests, so I'm taking a look.

File: docs/source/ruletypes.rst (hunk 1 above)
Comment (jertel) [reply docs/source/ruletypes.rst:2637]: I had intended this comment to be for the `iris_alert_context` line but GitHub expanded it to cover additional lines. So it's really iris_alert_context which needs a description. Sorry for the confusion.

**Comment by malinkinsa** [comment]
> The reason the automation failed is due to the linter finding some white space violations in your new code/tests.

Fixed

**Review by jertel — COMMENTED** [review]
This is looking good to me. I'll leave it open for a little while to let others offer their feedback.

**Review by jertel — APPROVED** [review]

**Comment by Areopagit** [comment]
IRIS alert context does not work correctly.
iris_alert_context:
hostname: host.hostname
domain: user.domain
ip: host.ip
user: winlog.event_data.TargetUserName
timestamp: event.created
message: message
log: winlog
tags: tags

hostname, domain, ip, user and timestamp dont work. This fields in IRIS are empty.
But field from a single word (message, winlog, tags) - OK!

The dot somehow gets in the way...

Please, help. All fields of Windows logs contain dot.

**Pull request merged**